torch/csrc/jit/python/pybind.h

```diff
@@ -17,6 +17,90 @@
 
 namespace py = pybind11;
 
+namespace torch {
+namespace jit {
+
+// This is a variant of shared_ptr that "sees through" a wrapper.
+// We use it to convert Value, Node, Block and node to "wrapped" Python
+// values. When we destruct the C++ object, the wrapper's pointer will
+// be set to 0 and any future dereferencing will throw. We need this
+// because the Python objects may hang around after the C++ object
+// has already been destroyed.
+// This also needs the magic type_caster below, which is from the
+// workaround offered in https://github.com/pybind/pybind11/issues/2751
+template <typename T>
```

Comment:
Should we add some static checks here to ensure T is one of the IR structs?

Reply:
Sounds like a good idea. Thank you!

Reply:
I added a static assert. Again, thank you for the suggestion!